src/core/processing/qgsprocessingprovider.cpp

```diff
@@ -107,6 +107,66 @@ QStringList QgsProcessingProvider::supportedOutputTableExtensions() const
   return supportedOutputVectorLayerExtensions();
 }
 
+bool QgsProcessingProvider::isSupportedOutputValue( const QVariant &outputValue, const QgsProcessingDestinationParameter *parameter, QgsProcessingContext &context, QString &error ) const
+{
+  QString outputPath = QgsProcessingParameters::parameterAsOutputLayer( parameter, outputValue, context );
+  error.clear();
+  if ( parameter->type() == QgsProcessingParameterVectorDestination::typeName()
+       ||  parameter->type() == QgsProcessingParameterFeatureSink::typeName() )
+  {
+    if ( outputPath.isEmpty() || outputPath.startsWith( QLatin1String( "memory:" ) ) )
+    {
+      if ( !supportsNonFileBasedOutput() )
+      {
+        error = tr( "This algorithm only supports disk-based outputs" );
+        return false;
+      }
+      return true;
+    }
+
+    QString providerKey;
+    QString uri;
+    QString layerName;
+    QMap<QString, QVariant> options;
+    bool useWriter = false;
+    QString format;
+    QString extension;
+    QgsProcessingUtils::parseDestinationString( outputPath, providerKey, uri, layerName, format, options, useWriter, extension );
+
+    if ( providerKey != QLatin1String( "ogr" ) )
```

Comment:
hmmmm... anything but OGR is a nonFileBased format?

Reply:
Yes - in processing, for outputs. The only choices are ogr, memory, or postgres.

Reply:
are we limitating any provider extending processing with custom alg managing custom formats... I suggest only to remove this if block and leave the control to the next call to supportedOutputVectorLayerExtensions

Reply:
I understand what you mean, but I'm certain other parts of processing won't cope with custom output formats (specifically modeller needs knowledge of all available output types, or it can't convert intermediate layers to the readable types for subsequent child algs).

Reply:
Yes, I supposed that was the reason, but in this way we are forcing to have provider without their custom ecosystem. BTW because I understand it's a the only 0.000001% of use cases => it does not make sense to add complexity for this.

Reply:
I don't think I explained the situation well - this code only applies to vector layer outputs. Algorithms are able to define their own custom output types using whatever format they want, but then they cease to be vector layer outputs (and this code no longer applies). E.g. the r provider could have an output parameter type for data frames, but that's **not** a vector layer output (since it can't be read as a normal vector layer in qgis). It's a custom RDataFrame output, and can't be piped into inputs expecting a vector layer for other algs in a model.

So vector layers have to be something qgis understands as a vector layer, which means a disk based format readable by ogr, PostGIS, (+ in future, MySQL and Oracle).

Reply:
well.. there are proprietary vector layers not supported by OGR. That's the reason was introduced pythin data providers.... vector layer for is an abstraction leaved to dataProvider it's managing. My concern is that there are companies that have already their strange formats and they want to integrate QGIS without changing other components.

Reply:
btw... sorry for delay, I'm not against it's merge